cli/flox-rust-sdk/src/providers/flox_cpp_utils.rs

```diff
@@ -155,13 +171,29 @@ impl InstallableLocker for Pkgdb {
             _ => FlakeInstallableError::Pkgdb(err),
         })?;
 
-        let lock = serde_json::from_value(lock)
+        let mut lock: LockedInstallable = serde_json::from_value(lock)
             .map_err(FlakeInstallableError::DeserializeLockedInstallable)?;
 
+        set_priority(&mut lock, descriptor);
+
         Ok(lock)
     }
 }
 
+/// Sets the priority for the locked installable.
+///
+/// The priority order of...the priority is:
+/// - Priority set in the descriptor
+/// - `meta.priority` of the derivation
+/// - Default priority
+fn set_priority(locked: &mut LockedInstallable, descriptor: &ManifestPackageDescriptorFlake) {
+    if descriptor.priority.is_some() {
+        locked.priority = descriptor.priority;
+    } else if locked.priority.is_none() {
+        locked.priority = Some(DEFAULT_PRIORITY);
```

Comment:
nonblocking question/note: I wonder if we should leave this as none and let buildenv decide. I think we set it for catalog to DEFAULT_PRIORITY as well though so I think this is consistent for now

Reply:
I think that's a separate change. Like you said, this is consistent with how we handle catalog packages and that was the goal for this change.

Reply:
Yep just a note